site/content/contribute/developer-setup.md

```diff
@@ -19,8 +19,20 @@ If you're migrating from before the monorepo see the [migration notes]({{< ref "
 If you're developing plugins, see the plugin [developer setup]({{< ref "/integrate/plugins/developer-setup" >}}) documentation.
 {{</note>}}
 
+# Prerequisites for WSL Development Environment
+
+1. [Install WSL](https://learn.microsoft.com/en-us/windows/wsl/install) by running the following command as an administrator in PowerShell: `wsl --install`
+2. [Install Docker Desktop for Windows](https://learn.microsoft.com/en-us/windows/wsl/tutorials/wsl-containers#install-docker-desktop) on your Windows machine.
+3. Perform the rest of the operations (except Docker installation) within the WSL environment and not in Windows.
+
 # Setup the Mattermost Server
 
+1. Install build essential tools:
+
+   ```sh
+   sudo apt install build-essential
```

Comment:
This is Ubuntu specific. If we want to add this detail we should call this out. @cwarnermm is there an existing pattern for this we want to use? So far this doc is pretty generic.

Reply:
Maybe we can just say here to install "make". And then probably give this as an example like on Ubuntu you can do this?

Reply:
Makes sense.